onnxruntime/test/onnx/TestCase.cc

```diff
@@ -366,7 +366,7 @@ TestModelInfo* TestModelInfo::LoadOnnxModel(_In_ const PATH_CHAR_TYPE* model_url
    * ???/output_??.pb
    */
 class OnnxTestCase : public ITestCase {
- private:
+ protected:
```

Comment:
Could you please put the new changes in 
```
#ifdef ENABLE_TRAINING
...
#endif
```
?

Reply:
It would make the really messy. We already run training ops like SoftmaxCrossEntropyLoss and NLLLoss as part of these test suites and don't special treatment for training. 

---
In reply to: [419620049](https://github.com/microsoft/onnxruntime/pull/3806#discussion_r419620049) [](ancestors = 419620049)

Reply:
Then why should we have one TestCase loader for every new op?